File: src/components/statics/faultPlot/processVector.py
import requests

filename = input("Enter filename: ").strip()
vectortype = input("Enter vectortype: ").strip()

if vectortype not in ["lines", "vector", "seismic", "stations"]:
    exit()

if "../" not in filename and "../../data/" not in filename:
    exit()

if vectortype == "lines":
    filecontents = requests.get(filename).text.split("X")
    tempArray = []
    
    error = 0
    for content in filecontents:
        lines = content.split("\n")
        tempArray.append(lines)
        if not lines[0].replace(".", "").isdigit():
            error += 1
        if not lines[1].replace(".", "").isdigit():
            error += 1
    
    if error == 0:
        print(tempArray)

File: src/components/statics/faultPlot/processVector.js
import { get } from 'request'

const filename = prompt('Enter filename:')
const vectortype = prompt('Enter vectortype:')

if (!['lines', 'vector', 'seismic', 'stations'].includes(vectortype)) {
    process.exit()
}

// if (!filename.includes('../../tesis/data/') && !filename.includes('../../data/')) {
//     process.exit()
// }

if (vectortype === 'lines') {
    get(filename, (error, response, body) => {
        const filecontents = body.split('X')
        const tempArray = []

        let errorCount = 0
        filecontents.forEach((content) => {
            const lines = content.split('\n')
            tempArray.push(lines)
            if (isNaN(parseFloat(lines[0]))) {
                errorCount++
            }
            if (isNaN(parseFloat(lines[1]))) {
                errorCount++
            }
        })

        if (errorCount === 0) {
            console.log(JSON.stringify(tempArray))
        }
    })
}
